index.js

```diff
@@ -2,19 +2,18 @@
 
 /* eslint-disable no-console, prefer-template */
 
-// The official way to annotate overloaded functions: https://github.com/facebook/flow/issues/60
-declare function printNodeType(node: *): string;
-// eslint-disable-next-line no-redeclare
-declare function printNodeType(node: *, branch: string): void;
-
-function printTree(
-  initialTree: *,
-  printNode: printNodeType,
-  getChildren: (node: *) => Array<*>,
-): void {
-  function printBranch(tree: *, branch: *): void {
+type PrintNode<T> = (node: T, branch: ?string) => ?string;
```

Comment:
Love the replacement of `*` with generics - but this isn't quite equivalent to my original flow statement

This would allow someone to pass in a function `(node, branch) => string` - but that's not valid - it should only allow either `(node, branch) => void)` or `(node) => string`.

Hence why I did the overloaded function.

Reply:
I think that's ok because it's taken care of here

https://github.com/BarryThePenguin/print-tree/blob/c0a19aa96ff1441c645c6aa2208c32c826fab151/index.js#L23-L27

Reply:
Yeah but if a consumer is using our flow-types they won't get a flow error now

Reply:
Those flow types weren't working though, that's why I changed it